File: api-server/internal/model/node.go
package model

import "time"

// Node 节点信息
type Node struct {
	NodeID        string     `gorm:"column:node_id;primaryKey" json:"nodeId"`
	HostID        *string    `gorm:"column:host_id" json:"hostId"`
	Hostname      *string    `gorm:"column:hostname" json:"hostname"`
	IPAddress     *string    `gorm:"column:ip_address" json:"ipAddress"`
	NPUCount      *int       `gorm:"column:npu_count" json:"npuCount"`
	Status        *string    `gorm:"column:status" json:"status"`
	LastHeartbeat *time.Time `gorm:"column:last_heartbeat" json:"lastHeartbeat"`
	CreatedAt     time.Time  `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt     time.Time  `gorm:"column:updated_at" json:"updatedAt"`
}

func (Node) TableName() string {
	return "nodes"
}

File: task-monitor-frontend/src/types/node.ts
// 节点类型定义
export interface Node {
  nodeId: string;
  hostId: string | null;
  hostname: string | null;
  ipAddress: string | null;
  npuCount: number | null;
  status: 'active' | 'inactive' | 'error' | null;
  lastHeartbeat: string | null;  // ISO 8601 格式
  createdAt: string;
  updatedAt: string;
}

export type NodeStatus = 'active' | 'inactive' | 'error';

export interface NodeListParams {
  status?: NodeStatus;
  page?: number;
  pageSize?: number;
}

export interface NodeListResponse {
  nodes: Node[];
  total: number;
  page: number;
  pageSize: number;
}

export interface NodeStats {
  total: number;
  active: number;
  inactive: number;
  error: number;
}
